R/prep_diagram.R

```diff
@@ -2,5 +2,67 @@ prep_diagram <- function(audit_data = NULL, emissions_data = NULL) {
   if (is.null(audit_data) | is.null(emissions_data)) {
     data_out <- use_toy_data("diagram")
   }
-  data_out
+
+  audit_data <- audit_data %>%
+    dplyr::mutate(exposure_portfolio = sum(.data$value_usd, na.rm = TRUE)) %>%
+    dplyr::group_by(.data$asset_type) %>%
+    dplyr::mutate(
+      exposure_asset_class = sum(.data$value_usd, na.rm = TRUE),
+      exposure_asset_class_perc = .data$exposure_asset_class / .data$exposure_portfolio
+    ) %>%
+    dplyr::ungroup() %>%
+    dplyr::mutate(
+      covered_pacta = dplyr::if_else(
+        .data$financial_sector %in% c("Automotive", "Coal", "Oil&Gas", "Power", "Aviation", "Cement", "Steel"),
+        TRUE,
+        FALSE
+      )
+    ) %>%
+    dplyr::group_by(
+      .data$asset_type, .data$exposure_portfolio, .data$covered_pacta,
+      .data$exposure_asset_class, .data$exposure_asset_class_perc
+    ) %>%
+    dplyr::summarise(
+      exposure_pacta = sum(.data$value_usd, na.rm = TRUE),
+      .groups = "drop"
+    ) %>%
+    dplyr::mutate(
+      exposure_pacta_perc_asset_class_exposure = .data$exposure_pacta / .data$exposure_asset_class
+    ) %>%
+    dplyr::ungroup() %>%
+    dplyr::filter(.data$covered_pacta) %>%
+    dplyr::select(
+      c(exposure_portfolio, asset_type, exposure_asset_class,
+        exposure_asset_class_perc, exposure_pacta, exposure_pacta_perc_asset_class_exposure)
+    )
+
+  emissions_data <- emissions_data %>%
+    dplyr::filter(.data$entity == "portfolio") %>%
+    dplyr::group_by(.data$asset_type) %>%
+    # TODO: check if this what we need or if it needs to be divivded by the asset value covered
+    dplyr::mutate(emissions_asset = sum(.data$weighted_sector_emissions, na.rm = TRUE)) %>%
+    dplyr::ungroup() %>%
+    dplyr::mutate(
+      covered_pacta = dplyr::if_else(
+        .data$sector %in% c("Automotive", "Coal", "Oil&Gas", "Power", "Aviation", "Cement", "Steel"),
```

Comment:
```suggestion
        .data$sector %in% c("Automotive", "Coal", "Oil&Gas", "Power", "Aviation", "Steel"),
```

Reply:
I think we're scrubbing Cement from everything, no?

Reply:
good question actually. so from the main parts looking at the portfolio in the executive summary: yes. From the overal pacta analysis, I think not? or did I miss that? In either case, the question would probably be if this flow chart should reflect only the sectors as covered in GECO 2021, since those are the ones used later on in the ES.

@Tilmon any thoughts on that?

Reply:
ok, well.... maybe not worth blocking this PR over... we can always remove it later

Reply:
I'll flag this with Tilman tomorrow morning and will handle it consistently across the repo